spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/keras/NewModule.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.keras
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+
+abstract class NewModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
+T: ClassTag](implicit ev: TensorNumeric[T]) extends AbstractModule[A, B, T] {
+
+  labor = null  // reset labor to be null
+
+  override def build(inputShape: Activity): Unit = {
+    labor = doBuild(inputShape)
+    super.build(inputShape)
+    this.output = labor.output
+    this.gradInput = labor.gradInput
+//    this.forwardTime = labor.forwardTime
+//    this.backwardTime = labor.backwardTime
+//    this.line = labor.line
+//    this.scaleB = labor.scaleB
+//    this.scaleBCache = labor.scaleBCache  //TODO: add this back and override all other methods
+//    this.scaleW = labor.scaleW
+//    this.scaleWCache = labor.scaleWCache
+//    this.train = labor.train
+  }
+
+  def doBuild(inputShape: Activity): AbstractModule[A, B, T]
+
+  // This method would only be called after `doBuilt`
+  override def doComputeOutputShape(inputShape: Activity): Activity = {
+    this.labor.doComputeOutputShape(inputShape)
+  }
+
+  override def accGradParameters(input: A, gradOutput: B): Unit = {
```

Comment:
The bad thing is we need to override lots of other methods in `AbstractModule` like this as well. cc @jason-dai